crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -805,20 +871,23 @@ impl LogsControllerInner {
         nodes_config: &NodesConfiguration,
         observed_cluster_state: &ObservedClusterState,
         effects: &mut Vec<Effect>,
+        node_set_selector_hints: impl NodeSetSelectorHints,
     ) {
         for (log_id, log_state) in &mut self.logs_state {
-            if log_state.try_transition_to_sealing(
+            if let ShouldSeal::Yes(new_configuration) = log_state.try_transition_to_sealing(
                 *log_id,
                 nodes_config,
                 self.current_logs.configuration(),
                 observed_cluster_state,
+                || node_set_selector_hints.preferred_sequencer(log_id),
             ) {
                 effects.push(Effect::Seal {
                     log_id: *log_id,
                     segment_index: log_state
                         .current_segment_index()
                         .expect("expect a valid segment"),
                     debounce: None,
+                    new_configuration,
```

Comment:
Not batching the individual seal and extend requests is not great because in some situation where all logs have to be sealed, we end up with many concurrent seal_and_extend requests that race each other. With the previous approach where we first seal and then reconfigure on the next tick, we had natural batching of extending multiple logs.